python/packages/nisar/products/writers/BaseL2WriterSingleInput.py

```diff
@@ -1506,6 +1507,70 @@ def populate_processing_information_l2_common(self):
                 output_ds_name_list=ds_name_list,
                 skip_if_not_present=True)
 
+    def geocode_isce3_lut(self, correction_lut, lut_name,
+                          timing_corrections_group_path, frequency):
+        '''
+        Geocode ISCE3 look-up table (LUT) object
+
+        Parameters
+        ----------
+        correction_lut: isce3.core.LUT2d
+            ISCE3 look-up table (LUT) object containing the data
+            to be geocoded
+        lut_name: str
+           Name of the LUT dataset in the output product metadata
+        timing_corrections_group_path: str
+            Path to the output HDF5 LUT group
+        frequency: str, optional
+            Frequency sub-band, used to read the sub-band wavelength.
+        '''
+
+        new_var_array = correction_lut.data
+
+        driver = gdal.GetDriverByName("GTiff")
+        dtype = gdal_array.NumericTypeCodeToGDALTypeCode(
+                new_var_array.dtype)
+        length, width = new_var_array.shape
+
+        scratch_path = self.cfg['product_path_group']['scratch_path']
+        temp_file = tempfile.NamedTemporaryFile(dir=scratch_path,
+                                                suffix='.tif')
+        dset = driver.Create(temp_file.name, xsize=width, ysize=length,
+                             bands=1, eType=dtype)
+        raster_band = dset.GetRasterBand(1)
+        raster_band.WriteArray(new_var_array)
+
+        # flush data to the disk (gc is the garbage collector)
+        raster_band.FlushCache()
+        del raster_band
+        del dset
+        gc.collect()
+
+        correction_raster = isce3.io.Raster(temp_file.name)
+
+        radar_grid_slc = self.input_product_obj.getRadarGrid(frequency)
+
+        radar_grid = isce3.product.RadarGridParameters(
+                correction_lut.y_start,
+                radar_grid_slc.wavelength,
+                1.0 / correction_lut.y_spacing,
+                correction_lut.x_start,
+                correction_lut.x_spacing,
+                radar_grid_slc.lookside,
+                correction_lut.length,
+                correction_lut.width,
+                radar_grid_slc.ref_epoch)
+
+        self.geocode_raster(correction_raster,
+                            timing_corrections_group_path,
+                            [lut_name],
+                            radar_grid,
+                            metadata_group='processingInformation',
+                            compute_stats=True,
+                            data_interpolator='nearest')
+
+        return frequency
```

Comment:
No need to return this. It's passed as an input. And we don't seem to be using the return value when we call this function in `populate_processing_information_timing_corrections()`.

```suggestion
```